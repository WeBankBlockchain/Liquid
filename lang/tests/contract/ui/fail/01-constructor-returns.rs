use liquid_lang as liquid;

#[liquid::contract]
mod noop {
    #[liquid(storage)]
    struct Noop {}

    #[liquid(methods)]
    impl Noop {
        pub fn new(&mut self) -> Self {}

        pub fn noop(&self, i: u8, s: String) -> (bool, i32) {}
    }
}

fn main() {}
