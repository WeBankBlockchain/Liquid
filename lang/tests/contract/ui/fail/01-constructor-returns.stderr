error: contract constructor should not have return value
  --> $DIR/01-constructor-returns.rs:10:31
   |
10 |         pub fn new(&mut self) -> Self {}
   |                               ^^^^^^^
